### PR TITLE
Updating icon-font element removing i

### PR DESCRIPTION
**Pull request by clrux**
A common bad pattern around the Web has been to use `i` for icon fonts. Semantically the `i` element meant italics, but since it's more correct to use `em`, the `i` has largely become unused except for icon fonts. This is a bad pattern nonetheless.

At edX we are moving away from the `i` element for icon fonts and instead using `span`. All markup and classes may remain the same; only the element changes.

This is follow-up work done by @ssemenova in the platform. It contains one update and one addition:

* updates terminology for previous icon-font checks to specify `span` rather than `i`
* adds new checks to ensure the use of `span` instead of `i` (will fail on `<i>` encounters)

Version bumps the package for use in platform.

## Reviewers

- [x] @benpatterson 

FYI @cptvitamin

### CODE REVIEW

**Comment by clrux** [comment]
Confirmed that this test works. I ran it on Master (using the commit hash in our package.json) and it didn't find any errors (thanks to @ssemenova). I manually included an incorrect usage on the LMS Dashboard to see if it could get caught, and sure enough!

File: lib/custom_a11y_rules.js [hunk 1]
```diff
@@ -192,5 +207,25 @@ var customRules = {
                 }
             },
         },
+        {
+            'id': 'icon-correct-element',
+            'metadata': {
+                'impact': 'minor',
+                'messages': {
+                    'pass': 'icon-fonts should use the i element, not span',
```
Comment (benpatterson) [inline lib/custom_a11y_rules.js]: Is this message correct? I thought we wanted folks to use span, not i...

Comment (clrux) [inline lib/custom_a11y_rules.js]: Yep, you're right. I've got it backwards.

**Comment by clrux** [comment]
@benpatterson Can you holla at this review?

**Comment by benpatterson** [comment]
👍

**Pull request merged**